gateway/ln-gateway/src/rpc/mod.rs

```diff
@@ -65,7 +68,8 @@ pub struct WithdrawPayload {
 pub struct FederationInfo {
     pub federation_id: FederationId,
     pub balance_msat: Amount,
-    pub config: ClientConfig,
+    #[serde(skip_serializing_if = "Option::is_none")] // Skip serializing if config is None
+    pub config: Option<ClientConfig>,
```

Comment:
Hmm so the backwards compatibility tests might be failing because this changed from `ClientConfig` to `Option<ClientConfig>` (havent confirmed myself yet).

I don't think changing this is actually necessary. The main issue with the `ClientConfig` is just that it clutters up the CLI response. I think we should still return `ClientConfig` on the `get_info` requests, but on the CLI just filter out the config according to the `exclude_config` flag. This will make it so the CLI is more readable but also keep the change simple on the API/server side.

Reply:
Oh perfect, sounds like a straightforward solution then. We will just update one file. 

I've made the changes and updated the PR. Can you please review it again and run the jobs to check if the backward compatibility issue still occurs?